### PR TITLE
Makes admins not count towards the playercount cap

### PR DESCRIPTION
**Pull request by deathride58**
## About the PR
This is a straight-forward PR, making it so that admins don't count towards the playercount cap. This is controlled by the new `admin.admins_count_for_max_players` cvar

## Why / Balance
Currently, admins bypass the playercount cap entirely, meaning it's very common to see the servers end up escalating beyond their actual playercount cap. This means that when an admin joins while the server's at the cap, players have to wait for way longer than necessary for more than just one person to leave if they want to join the server, which is immensely frustrating for players. Simple fix: making it so that those who are able to bypass the playercount cap don't count towards it either.

## Technical details
This adds a new CVar controlling whether admins count towards the playercount cap (if true, then it behaves exactly how the playercount cap behaved before).

When that CVar is false, the number of active admins gets subtracted from the current playercount during connection checks that care about the playercount.

## Media
N/A

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
N/A

**Changelog**

:cl: Bhijn and Myr
- tweak: Admins no longer count towards the playercount cap, meaning you no longer need to wait for multiple people to leave in order to join after an admin has joined.

### CODE REVIEW

File: Content.Shared/CCVar/CCVars.Admin.cs [hunk 1]
```diff
@@ -149,6 +149,15 @@ public sealed partial class CCVars
     public static readonly CVarDef<bool> AdminBypassMaxPlayers =
         CVarDef.Create("admin.bypass_max_players", true, CVar.SERVERONLY);
 
+    /// <summary>
+    ///     Determines whether admins count towards the total playercount when determining whether the server is over <see cref="SoftMaxPlayers"/>
+    ///     Ideally this should be used in conjuction with <see cref="AdminBypassPlayers"/>.
+	///     ///     This also applies to playercount limits in whitelist conditions
```
[the remark this answers is not in the document]
Comment (deathride58) [reply Content.Shared/CCVar/CCVars.Admin.cs:155]: Vscode moment

**Comment by SaphireLattice** [comment]
This will still show "81/80" if there's two admins on, but will let one person join, if I read this right?

That has to be pretty confusing. It should remove them from the reported player count.

**Comment by ilovehans10** [comment]
> This will still show "81/80" if there's two admins on, but will let one person join, if I read this right?
> 
> That has to be pretty confusing. It should remove them from the reported player count.

Honestly, it would be nice if there was an indicator for how many admins were online, and it could be separate from the player count still. Something where when you opened the dropdown for a server, it would have the server description, and then the number of admins online. That was this godsend can be implemented, and I can finally have a chance to hope on Harmony when there are 4 admins online (like I'm waiting for as I write this).\

Current:
Screenshot from 2024-11-23 13-29-29
Concept:
Admins online
(GIMP FTW)

**Comment by Aeshus** [comment]
> > This will still show "81/80" if there's two admins on, but will let one person join, if I read this right?
> > That has to be pretty confusing. It should remove them from the reported player count.
> 
> Honestly, it would be nice if there was an indicator for how many admins were online, and it could be separate from the player count still. Something where when you opened the dropdown for a server, it would have the server description, and then the number of admins online. That was this godsend can be implemented, and I can finally have a chance to hope on Harmony when there are 4 admins online (like I'm waiting for as I write this).\
> 
> Current: Screenshot from 2024-11-23 13-29-29 Concept: Admins online (GIMP FTW)

Wouldn't this also be a signal to raiders that they should go on servers with few admins online?

**Comment by ArtisticRoomba** [comment]
most servers desire concealing the presence of admins
if raiders have an easy way to check if admins are online, it promotes people raiding at times when admins are offline
this can also extend to regular players deciding to skirt the rules or violate them because they saw that there are zero admins connected

making admins not count towards the playercap would actually be beneficial to admins as players could no longer half-guess when admins are on a server (a server at 81/80 players means that there *has* to be at least one admin online).

**Comment by ilovehans10** [comment]
> most servers desire concealing the presence of admins if raiders have an easy way to check if admins are online, it promotes people raiding at times when admins are offline this can also extend to regular players deciding to skirt the rules or violate them because they saw that there are zero admins connected
> 
> making admins not count towards the playercap would actually be beneficial to admins as players could no longer half-guess when admins are on a server (a server at 81/80 players means that there _has_ to be at least one admin online).

That totally makes sense, but why not just also make it a server option then. If you ask me with this option enabled it shouldn't show the admins in the player count, otherwise people won't know if there are spaces open once the server hits the cap. For example if a server is at 50/50 players online, but 5 of those players are admin and as such don't count towards the player count, most people would assume that the server is full even though it isn't.

The most server owner friendly option would probably be to provide a cvar for whether admins should be counted as players for the player count in the launcher.

**Comment by Errant-4** [comment]
> Resolves #33424 I think

What issue where you trying to link? That's _this_ PR

**Pull request merged**